src/Accounts/Authentication/Factories/AuthenticationFactory.cs

```diff
@@ -94,79 +94,46 @@ public IServicePrincipalKeyStore KeyStore
             IAzureTokenCache tokenCache,
             string resourceId = AzureEnvironment.Endpoint.ActiveDirectoryServiceEndpointResourceId)
         {
-            IAccessToken token;
-            var cache = tokenCache as TokenCache;
+            IAccessToken token = null;
+            var cache = tokenCache.GetUserCache() as TokenCache;
             if (cache == null)
             {
-                cache = TokenCache.DefaultShared;
+                cache = new TokenCache();
             }
 
-            Task<IAccessToken> authToken;
-            if (Builder.Authenticator.TryAuthenticate(account, environment, tenant, password, promptBehavior, Task.FromResult(promptAction), tokenCache, resourceId, out authToken))
+            AuthenticationClientFactory authenticationClientFactory;
+            if (!AzureSession.Instance.TryGetComponent(AuthenticationClientFactory.AuthenticationClientFactoryKey, out authenticationClientFactory))
             {
-                return authToken.ConfigureAwait(false).GetAwaiter().GetResult();
+                throw new NullReferenceException(Resources.AuthenticationClientFactoryNotRegistered);
             }
 
-            var configuration = GetAdalConfiguration(environment, tenant, resourceId, cache);
-
-            TracingAdapter.Information(
-                Resources.AdalAuthConfigurationTrace,
-                configuration.AdDomain,
-                configuration.AdEndpoint,
-                configuration.ClientId,
-                configuration.ClientRedirectUri,
-                configuration.ResourceClientUri,
-                configuration.ValidateAuthority);
-            if (account != null && account.Type == AzureAccount.AccountType.ManagedService)
-            {
-                token = GetManagedServiceToken(account, environment, tenant, resourceId);
-            }
-            else if (account != null && environment != null
-                && account.Type == AzureAccount.AccountType.AccessToken)
+            Task<IAccessToken> authToken;
```

Comment:
I am worried that we are not doing anough tracing here.  When there is an authentication error, it is useful to know what values were passed to the authentication library, and what the results were - without this, pinpointing authentication errors can be difficult